code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/file/docker/OpenedDockerComposeFile.xtend

```diff
@@ -83,145 +66,102 @@ class OpenedDockerComposeFile {
     def addOrReplaceDockerComposePart(IntermediateOperationNode node) {
         switch (node) {
             IntermediateInfrastructureNode : {
-                val serv = '''«getServiceFromNode(node)»«"\n"»'''
-                addOrReplace(node.name, serv)
+                addOrReplace(getServiceFromNode(node))
             }
             IntermediateContainer: {
-                val serv = '''«getServiceFromContainer(node)»«"\n"»'''
-                addOrReplace(node.name, serv)
+                addOrReplace(getServiceFromContainer(node))
             }
             default : throw new IllegalArgumentException("OperationNode instance not supported.")
         }
     }
 
     /**
-     * Create a default network for the docker compose file. This is needed because LEMMA's
-     * operation modeling language does not support network configurations at the moment.
+     * Add or replace a docker-compose part to the docker-compose file based on the name of the
+     * operation node.The configuration of the docker-compose part is created by an intermediate
+     * deployment aspect.
+     */
+    def addOrReplaceDockerComposePart(String name, String dockerComposeServicePart) {
+        val adjustedDockerPart = Util::adjustIndentations("", dockerComposeServicePart) + "\n"
+        val mapper = new ObjectMapper(new YAMLFactory());
+        val service = mapper.readValue(adjustedDockerPart, DockerComposeService)
+        dockerComposeFile.services.put(name.toLowerCase, service)
+    }
+
+    /**
+     * Create a new docker compose file and add ne default network. The default network is needed
+     * because LEMMA's operation modeling language does not support network configurations at the
+     * moment.
      */
-    private def createDefaultNetwork() {
-        val network = new Network
+    private def createNewDockerFile() {
+        dockerComposeFile = new DockerComposeFile
+        var network = new DockerComposeNetwork()
         network.driver = "bridge"
-        dockerComposeNetworkParts.add(
-            DockerComposeTemplate::buildDockerComposeNetwork("default-network", network)
-        )
+        dockerComposeFile.networks.put("default-network", network)
+        dockerComposeFile.version = "3.7"
     }
 
     /**
      * Add or replace the docker-compose parts based the the name of the intermediate operation
      * node.
      */
-    private def addOrReplace(String name, String dockerComposeServicePart) {
-        dockerComposeServiceParts.put(name.toLowerCase, dockerComposeServicePart)
+    private def addOrReplace(DockerComposeService service) {
```

Comment:
How often is this method called? If only in one or two places, you might just replace its call with the invocation of `put`, because `put` semantics are essentially "add or replace". So probably no need for a dedicated method.